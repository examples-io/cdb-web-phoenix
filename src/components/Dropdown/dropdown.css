select.input--select {
    min-width: 7rem;
    border-bottom: none;
}