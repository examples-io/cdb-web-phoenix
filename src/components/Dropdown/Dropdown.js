import React, { Component } from 'react'
import PropTypes from 'prop-types'

import './dropdown.css'

/**
 * @description Dropdown component - renders a dropdown selector with a label
 */
class Dropdown extends Component {
    render() {
        return (
            <div className="form--input">
                <label className="label" htmlFor={this.props.id} > {this.props.label} </label>
                <select className="input--select" id={this.props.id}
                        name={this.props.name}
                        onChange={this.props.onChange}
                        value={this.props.value} >
                    {this.renderOptions(this.props.options)}
                </select>
            </div>
        );
    }

    /**
     * @description renders a list of options to be nested in the <select> element
     * @param options {array} list of options to be rendered
     */
    renderOptions(options) {
        return options.map((option) => {
            return <option value={option.id} key={option.id} > {option.text} </option>;
        });
    }
}

Dropdown.propTypes = {
    /**
     * options - array of options to be rendered in the dropdown
     */
    options: PropTypes.array,
    /**
     * id - element id of the dropdown
     */
    id: PropTypes.string,
    /**
     * onChange - function to be attached to the dropdown's onChange event
     */
    onChange: PropTypes.func,
    /**
     * name - element name of the dropdown
     */
    name: PropTypes.string,
    /**
     * label - text to be displayed in the label
     */
    label: PropTypes.string
};

export default Dropdown;