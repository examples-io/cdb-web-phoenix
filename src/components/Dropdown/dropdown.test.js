import React from 'react';
import renderer from 'react-test-renderer';

import Dropdown from './Dropdown';

describe('Dropdown tests', () => {
    it ('renders Dropdown correctly', () => {
        const options = [
            { id: 'opt1', text: 'option 1' },
            { id: 'opt2', text: 'option 2' },
            { id: 'opt3', text: 'option 3' }
        ];
        const tree = renderer
            .create(<Dropdown id="testDropdown" name="test" label="Test Dropdown:" options={options} />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});