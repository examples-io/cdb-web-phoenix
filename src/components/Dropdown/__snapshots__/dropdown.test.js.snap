// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown tests renders Dropdown correctly 1`] = `
<div
  className="form--input"
>
  <label
    className="label"
    htmlFor="testDropdown"
  >
     
    Test Dropdown:
     
  </label>
  <select
    className="input--select"
    id="testDropdown"
    name="test"
    onChange={undefined}
    value={undefined}
  >
    <option
      value="opt1"
    >
       
      option 1
       
    </option>
    <option
      value="opt2"
    >
       
      option 2
       
    </option>
    <option
      value="opt3"
    >
       
      option 3
       
    </option>
  </select>
</div>
`;
